configure.ac

```diff
@@ -421,16 +425,23 @@ AS_IF([test "x$with_readline" != xno],[
 
   dnl We check for the symbol rl_ding to distinguish GNU readline
   dnl from other readline implementation.
-  AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline], [rl_ding], [], [], [])
-
+  case "$ac_cv_search_rl_bind_keyseq" in
+     *ereadline)
+       AX_CHECK_LIBRARY([READLINE], [ereadline/readline/readline.h], [ereadline], [rl_ding], [], [], [])
+       ;;
+     *)
+       AX_CHECK_LIBRARY([READLINE], [readline/readline.h], [readline],  [rl_ding], [], [], [])
+       ;;
+  esac
   AS_IF([test $ax_cv_have_READLINE = yes],[
     AC_DEFINE([HAVE_LIBREADLINE], [1], [Define if you have libreadline])
   ],[
     AS_CASE([x"$with_readline"],
       [x], [READLINE_CPPFLAGS= READLINE_LDFLAGS= READLINE_LIBS=],
-      [*], [AC_MSG_FAILURE([--with-readline was given, but test for readline failed])]
+      [*], [AC_MSG_FAILURE([--with-readline was given, but further tests for readline failed])]
     )
   ])
+ ], [AC_MSG_FAILURE([--with-readline was given, but first test for readline failed])])
```

Comment:
Hmm, is that true? Looking at the code now, it seems to me this check also is run if `--with-readline` is not given; the only way to bypass it is an explicit `--without-readline` resp. `--with-readline=no`. So that would mean we now force everybody who does not have GNU readline installed to pass one of those explicitly, which is not what we want.

But perhaps I am misreading the code -- have you tested this scenario?

Reply:
oops, I've forgotten to add the condition, to bail out only if readline is requested. Fixed by latest commit.